pyomo/contrib/mindtpy/algorithm_base_class.py

```diff
@@ -1100,8 +1100,9 @@ def solve_subproblem(self):
                 tolerance=config.constraint_tolerance,
             )
         except InfeasibleConstraintException as e:
+            config.logger.error(e, exc_info=True)
             config.logger.error(
-                str(e) + '\nInfeasibility detected in deactivate_trivial_constraints.'
+                'Infeasibility detected in deactivate_trivial_constraints.'
```

Comment:
In this pattern, it is still good to include the text from the original error. Here you are hiding it from the user completely and replacing it with your own.